Volume/Networking/Network.swift

```diff
@@ -87,3 +87,16 @@ struct GraphQLPublisher<Query: GraphQLQuery>: Publisher {
         subscriber.receive(subscription: subscription)
     }
 }
+
+class NetworkState: ObservableObject {
+    @Published var networkFailed = false
+
+    func handleNetworkFailure(_ completion: Subscribers.Completion<WrappedGraphQLError>) {
```

Comment:
Instead of adding network handling code in every single `sink`, why not do this updating logic in `fetchQuery`?

Reply:
That was actually my original approach but I couldn't find a way to do this.  I can revisit this idea though because I definitely thought this was the most scalable approach